daemon/restart.go

```diff
@@ -39,8 +39,10 @@ func (daemon *Daemon) containerRestart(container *container.Container, seconds i
 		return err
 	}
 
-	if err := daemon.containerStart(container); err != nil {
-		return err
+	if !container.HostConfig.AutoRemove {
```

Comment:
I'm not sure about this one. I think we should allow the client to restart a container w/o it being removed. Granted this means that "docker stop" and "docker start" will not be the exact same as "docker restart" in this case but I think this is what the user would expect - otherwise they wouldn't be trying to do "docker restart" at all.